c/str/split_into_nhot.cc

```diff
@@ -0,0 +1,159 @@
+//------------------------------------------------------------------------------
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// © H2O.ai 2018
+//------------------------------------------------------------------------------
+#include "str/py_str.h"
+#include <unordered_map>
+#include <vector>
+#include "datatable.h"
+#include "utils/exceptions.h"
+#include "utils/omp.h"
+#include "utils/shared_mutex.h"
+
+namespace dt {
+
+/**
+ * Split string into tokens
+ */
+static void tokenize_string(
+    std::vector<std::string>& tokens,
+    const char* strstart,
+    const char* strend,
+    char sep
+) {
+  tokens.clear();
+  const char* ch = strstart;
+  while (ch < strend) {
+    if (*ch == ' ' || *ch == '\t' || *ch == '\n' || *ch == sep) {
+      ch++;
+    }
+    else {
+      const char* ch0 = ch;
+      // Handle the case of quoted token
+      if (*ch == '\'' || *ch == '"') {
+        char quote = *ch++;
+        while (ch < strend && *ch != quote) {
+          ch += 1 + (*ch == '\\');
+        }
+        if (ch < strend) {
+          tokens.emplace_back(ch0 + 1, ch);
+          ch++;  // move over the closing quote
+          continue;
+        }
+        // fall-through
+        ch = ch0;
+      }
+      // Regular non-quoted token, parse until the next sep
+      while (ch < strend && *ch != sep) {
+        ch++;
+      }
+      const char* ch1 = ch;
+      while (ch1 > ch0) {  // also omit trailing spaces
+        char c = *(ch1 - 1);
+        if (!(c == ' ' || c == '\t' || c == '\n')) break;
+        ch1--;
+      }
+      tokens.emplace_back(ch0, ch1);
+      ch++;  // move over the sep
+    }
+  }
+}
+
+
+
+DataTable* split_into_nhot(Column* col, char sep) {
+  bool is32 = (col->stype() == SType::STR32);
+  bool is64 = (col->stype() == SType::STR64);
+  xassert(is32 || is64);
+  const uint32_t* offsets32;
+  const uint64_t* offsets64;
+  const char* strdata;
+  if (is32) {
+    auto scol = static_cast<StringColumn<uint32_t>*>(col);
+    offsets32 = scol->offsets();
+    strdata = scol->strdata();
+  } else {
+    auto scol = static_cast<StringColumn<uint64_t>*>(col);
+    offsets64 = scol->offsets();
+    strdata = scol->strdata();
+  }
+
+  size_t nrows = static_cast<size_t>(col->nrows);
+  std::unordered_map<std::string, size_t> colsmap;
+  std::vector<Column*> outcols;
+  std::vector<int8_t*> outdata;
+  std::vector<std::string> outnames;
+  dt::shared_mutex shmutex;
+
+  OmpExceptionManager oem;
+  #pragma omp parallel
```

Comment:
What about doing `#pragma omp parallel num_threads(nth0)` here? `nth0` can be `config::nthreads` it its simplest form, or 
```
int nth0 = std::min(config::nthreads,
                        static_cast<int>(nrows / min_nrows_per_thread));
```
like we do in other places.